### PR TITLE
Feature/421 keybinding refactoring

### PR DESCRIPTION
**Pull request by danielsteinkogler**
## Basics

- [ ] I added a line to [changelog.md](/doc/changelog.md)
- [ ] The PR is rebased with current master.
- [ ] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages and changelog.
- [ ] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added automated tests or a [manual test protocol](../doc/tests/manual/protocol.md)
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)
- [ ] Exceptions to any guidelines are documented

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] I've read through the whole documentation
- [ ] I've checked conformity to guidelines

### CODE REVIEW

File: frontend/src/config/keybindings/index.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * @param keyHandlerActions - An object that maps action names to corresponding handler functions.
+ *
+ * @returns A record of key event handlers, where each key represents a keyboard shortcut defined in the
+ *          scope's configuration, and the corresponding value is the handler function associated with the
+ *          action triggered by that shortcut. If no matching keybindings are found for the given scope,
+ *          an empty object is returned.
+ */
+export function createKeyBindingsAccordingToConfig(
+  scope: string,
+  keyHandlerActions: Record<string, () => void>,
```
Comment (markus2330) [inline frontend/src/config/keybindings/index.ts:30]: Also rename here to keyBindingsActions

File: frontend/src/config/keybindings/index.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+export const keyBindingConfig: KeyBindingConfig = keybindings;
+
+/**
+ * Creates a set of key event handlers based on the provided scope and a set of key handler actions.
+ * These handlers are mapped to specific keyboard shortcuts defined in the configuration for the given scope.
+ *
+ * @param scope - The scope for which to create key event handlers.
+ * @param keyHandlerActions - An object that maps action names to corresponding handler functions.
```
Comment (markus2330) [inline frontend/src/config/keybindings/index.ts:21]: Also rename here. Write that it is an array.

File: frontend/src/config/keybindings/index.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import keybindings from './keybindings.json';
+import { createShortcutIncludingModifierKeysFromKeyEvent } from '@/utils/key-combinations';
+import React from 'react';
+
+export const KEYBINDING_SCOPE_GLOBAL = 'global';
+export const KEYBINDING_SCOPE_PLANTS_LAYER = 'plants_layer';
```
Comment (markus2330) [inline frontend/src/config/keybindings/index.ts:6]: add all the layers

File: frontend/src/config/keybindings/index.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+type KeyBinding = Record<string, string[]>;
+
+type KeyBindingConfig = {
```
Comment (markus2330) [inline frontend/src/config/keybindings/index.ts:10]: consistently plural?

```suggestion
type KeyBindingsConfig = {
```

**Review by markus2330 — CHANGES_REQUESTED** [review]
Great job, looks good to me. Only a few fixes to be consistent with terms in the concept.

**Comment by danielsteinkogler** [comment]
@markus2330 i think we can merge this after approval

**Review by markus2330 — CHANGES_REQUESTED** [review]
Needs rebase. Could you please add keyhandler and keybinding to glossary.

**Comment by danielsteinkogler** [comment]
@markus2330 i rebased and added the terms to glossary. Please check if the description is okay for you.

**Review by markus2330 — APPROVED** [review]
Great, much clearer.

**Pull request merged**